### PR TITLE
openshift/hypershift: update e2e job for BYO infra

### PR DESCRIPTION
**Pull request by csrwng**
Add steps for creation and teardown of infra around e2e test

### CODE REVIEW

File: ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml [hunk 1]
```diff
@@ -46,6 +46,40 @@ tests:
 - as: e2e-aws
   steps:
     cluster_profile: aws
+    post:
+    - as: cleanup-infra
+      cli: latest
+      commands: |-
+        bin/hypershift destroy infra aws \
+          --infra-id $(cat "${SHARED_DIR}/infra-id")
+          --aws-creds "${CLUSTER_PROFILE_DIR}/.awscred" && \
```
Comment (ironcladlou) [inline ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml:55]: Should we run `destroy iam` regardless of the outcome of infra? We only get one chance to execute

**Comment by csrwng** [comment]
@ironcladlou this should be ready, the rehearse job passed 🎉

**Comment by ironcladlou** [comment]
Thanks!
/lgtm

**Pull request merged**